scripts/generate-sitemap.js

```diff
@@ -53,10 +67,39 @@ fs.readFile('./public/sitemap.xml', 'utf8', async (err, data) => {
       })
     }
 
+    const nftResponse = await fetch('https://api.uniswap.org/v1/graphql', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Origin: 'https://app.uniswap.org',
+      },
+      body: JSON.stringify({ query: nftQuery }),
+    })
+    const nftJSON = await nftResponse.json()
+    const collectionAddresses = nftJSON.data.topCollections.edges.map((edge) => edge.node.nftContracts[0].address)
+    collectionAddresses.forEach((address) => {
+      const collectionURL = `https://app.uniswap.org/nfts/collection/${address}`
+      if (!(collectionURL in sitemapURLs)) {
+        sitemap.urlset.url.push({
+          loc: [collectionURL],
+          lastmod: [nowISO],
+          priority: [0.7],
+        })
+      }
+    })
+
     const builder = new Builder()
     const xml = builder.buildObject(sitemap)
-    fs.writeFile('./public/sitemap.xml', xml, (error) => {
+    const path = './public/sitemap.xml'
+    fs.writeFile(path, xml, (error) => {
       if (error) throw error
+      const stats = fs.statSync(path)
+      const fileSizeBytes = stats.size
+      const fileSizeMegabytes = fileSizeBytes / (1024 * 1024)
+
+      if (fileSizeMegabytes > 50) {
+        throw new Error('File size exceeds 50MB')
```

Comment:
nit: to help oncall when tracking down the issue. Should we also add the `sitemap` to help with debugging?
```suggestion
        throw new Error('Generated sitemap file size exceeds 50MB')
```